engine/src/main/java/org/terasology/rendering/nui/contextMenu/MenuTree.java

```diff
@@ -20,26 +20,50 @@
 import java.util.Map;
 import java.util.function.Consumer;
 
-public class ContextMenuTree {
-    private Map<String, ContextMenuOption> options = Maps.newHashMap();
-    private Map<String, ContextMenuTree> submenues = Maps.newHashMap();
-    public boolean visible = false;
+/**
+ * A data structure to store {@link ContextMenuScreen} options and submenues.
+ */
+public class MenuTree {
+    /**
+     * The options of this menu.
+     */
+    private Map<String, AbstractContextMenuItem> options = Maps.newLinkedHashMap();
+
+    /**
+     * The submenues of this menu.
+     */
+    private Map<String, MenuTree> submenues = Maps.newLinkedHashMap();
```

Comment:
I meant both maps to be replaced by a single `List<AbstractContextMenuItem>` where both MenuTree and MenuOption extend from it.

MenuTree would then contain no logic at all (no visiblity setting).